src/sagas/reown.js

```diff
@@ -0,0 +1,861 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {
+  call,
+  fork,
+  take,
+  all,
+  put,
+  cancelled,
+  takeLatest,
+  takeEvery,
+  select,
+  race,
+  actionChannel,
+} from 'redux-saga/effects';
+import { eventChannel } from 'redux-saga';
+import { get, values } from 'lodash';
+import { Core } from '@walletconnect/core';
+import { WalletKit } from '@reown/walletkit';
+import {
+  TriggerTypes,
+  TriggerResponseTypes,
+  RpcResponseTypes,
+  handleRpcRequest,
+  CreateTokenError,
+  SendNanoContractTxError,
+} from '@hathor/hathor-rpc-handler';
+import { isWalletServiceEnabled } from './wallet';
+import { ReownModalTypes } from '../components/Reown/ReownModal';
+import {
+  REOWN_PROJECT_ID,
+  REOWN_FEATURE_TOGGLE,
+} from '../constants';
+import {
+  types,
+  setReownSessions,
+  onExceptionCaptured,
+  setWCConnectionFailed,
+  setNewNanoContractStatusLoading,
+  setNewNanoContractStatusReady,
+  setNewNanoContractStatusFailure,
+  setNewNanoContractStatusSuccess,
+  setCreateTokenStatusLoading,
+  setCreateTokenStatusReady,
+  setCreateTokenStatusSuccessful,
+  setCreateTokenStatusFailed,
+  showGlobalModal,
+  hideGlobalModal,
+} from '../actions';
+import { checkForFeatureFlag, getNetworkSettings, retryHandler, showPinScreenForResult } from './helpers';
+import { logger } from '../utils/logger';
+import { getGlobalReown, setGlobalReown } from '../modules/reown';
+import { MODAL_TYPES } from '../components/GlobalModal';
+import { getGlobalWallet } from '../modules/wallet';
+
+const log = logger('reown');
+
+const AVAILABLE_METHODS = {
+  HATHOR_SIGN_MESSAGE: 'htr_signWithAddress',
+  HATHOR_SEND_NANO_TX: 'htr_sendNanoContractTx',
+  HATHOR_SIGN_ORACLE_DATA: 'htr_signOracleData',
+  HATHOR_CREATE_TOKEN: 'htr_createToken',
+};
+
+const AVAILABLE_EVENTS = [];
+
+const ERROR_CODES = {
+  UNAUTHORIZED_METHODS: 3001,
+  USER_DISCONNECTED: 6000,
+  USER_REJECTED: 5000,
+  USER_REJECTED_METHOD: 5002,
+  INVALID_PAYLOAD: 5003,
+};
+
+function* isReownEnabled() {
+  const reownEnabled = yield call(checkForFeatureFlag, REOWN_FEATURE_TOGGLE);
+  return reownEnabled;
+}
+
+function* init() {
+  log.debug('Wallet not ready yet, waiting for START_WALLET_SUCCESS.');
+  yield take(types.START_WALLET_SUCCESS);
+
+  // We should check if nano contracts are enabled in this network:
+  const nanoContractsEnabled = yield select((state) => get(state.serverInfo, 'nanoContractsEnabled', false));
+  if (!nanoContractsEnabled) {
+    log.debug('Nano contracts are not enabled, skipping reown init.');
+    return;
+  }
+
+  try {
+    const walletServiceEnabled = yield call(isWalletServiceEnabled);
+    const reownEnabled = yield call(isReownEnabled);
+
+    if (walletServiceEnabled) {
+      log.debug('Wallet Service enabled, skipping reown init.');
+      return;
+    }
```

Comment:
Shouldn't it be the opposite? If reown is enabled, we disable wallet service.

Reply:
Why?

Reply:
Right now we have the wallet service enabled for all the users in mainnet. If we want to do the reown rollout progressively, we would need to remove all users from wallet service.

Reply:
This is the wallet-desktop, we have 0% rollout